vllm/config.py

```diff
@@ -1721,18 +1722,13 @@ def _get_and_verify_max_len(
             default_max_len)
         derived_max_model_len = default_max_len
 
+    # Backwards compatibility
+    patch_rope_scaling(hf_config)
+
```

Comment:
Why are we patching this twice? Isn't `hf_config` already patched inside `get_config`?

I also think it's probably better to keep the patching here versus inside `get_config` - That was we keep the original hf config object intact. WDYT?

Reply:
Originally I thought so too, but turns out that some of the tests pass in outdated keys to this function directly.

Reply:
So either we have to update the tests (which I'm not familiar with) or patch it one more time.

Reply:
Let's only patch for rope scaling here once then inspect which tests are failing. IMO we should keep tests serve for codebase, not the other way around.

Reply:
Alright, I'll update the tests then.

Reply:
Done, PTAL again.